pkg/controllers/termination/eviction.go

```diff
@@ -0,0 +1,87 @@
+package termination
+
+import (
+	"context"
+	"sync"
+
+	set "github.com/deckarep/golang-set"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type EvictionQueue struct {
+	queue        workqueue.RateLimitingInterface
+	coreV1Client corev1.CoreV1Interface
+
+	enqueued set.Set
+	once     sync.Once
+}
+
+// Evict adds a pod to the EvictionQueue
+func (e *EvictionQueue) Add(pods []*v1.Pod) {
+	// Start processing eviction queue if it hasn't started already
+	e.once.Do(func() { go e.run() })
+
+	for _, pod := range pods {
+		nn := types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}
+		if !e.enqueued.Contains(nn) {
+			e.enqueued.Add(nn)
+			e.queue.Add(nn)
+		}
+	}
+}
+
+func (e *EvictionQueue) run() {
+	for {
+		// Get pod from queue. This waits until queue is non-empty.
+		item, shutdown := e.queue.Get()
+		if shutdown {
+			zap.S().Infof("Shutdown queue is broken")
+		}
+		// Panic if not a NamespacedName
+		nn := item.(types.NamespacedName)
+		// Evict pod
+		evicted := e.evict(nn)
+		e.queue.Done(nn)
+		if evicted {
+			zap.S().Debugf("Successfully evicted pod %s", nn.String())
```

Comment:
I'm trying to get away from the "Successfully" log lines, since it's typically already captured by the log type (i.e., we can assume success if its not errorf). I might simply say "Evicted pod %s"